src/Microsoft.ML.StandardTrainers/StandardTrainersCatalog.cs

```diff
@@ -487,6 +494,13 @@ public IClassificationLoss CreateComponent(IHostEnvironment env)
         /// </summary>
         /// <param name="catalog">The regression catalog trainer object.</param>
         /// <param name="options">Trainer options.</param>
+        /// <example>
+        /// <format type="text/markdown">
+        /// <![CDATA[
+        /// [!code-csharp[OnlineGradientDescent](~/../docs/samples/docs/samples/Microsoft.ML.Samples/Dynamic/Trainers/Regression/OnlineGradientDescentWithOptions.cs)]
+        /// ]]>
```

Comment:
also needs #2425 first

Reply:
#2425 is independent to this PR, right? Goal of this PR is building template. Afterwards, we can still do what we want.

---
In reply to: [275564113](https://github.com/dotnet/machinelearning/pull/3319#discussion_r275564113) [](ancestors = 275564113)

Reply:
I agree with Shahab. Have documentation which states "this learner is broken here is tracker number" is worse than no documentation at all.
I would probably even make OGD internal until we figure out why it's so bad.

---
In reply to: [275592181](https://github.com/dotnet/machinelearning/pull/3319#discussion_r275592181) [](ancestors = 275592181,275564113)